### PR TITLE
Add support for multiple SourceLine tags (with build fixes)

### PR DESCRIPTION
**Pull request by mchenryc**
This PR builds on [jakobdoc](https://github.com/jakobdoc)'s fix for Multiple SourceLines https://github.com/jwgmeligmeyling/spotbugs-github-action/pull/17. That PR has developed build issues, so here I have merged changes from upstream, and fix a couple minor linting issues.

Note this should also fix most (if not all) of the dependabot PRs.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ inputs:
       [Learn more about creating and using encrypted secrets](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets)
     default: ${{ github.token }}
 runs:
-  using: 'node12'
+  using: 'node16'
```
Comment (jwgmeligmeyling) [inline action.yml:24]: Any idea how to work around `Error: System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter ''using: node16' is not supported, use 'docker' or 'node12' instead.')` in the  CI test?

Comment (mchenryc) [reply action.yml:24]: Seems github runners only support node12. Seems we either need to create docker container, or investigate https://github.com/actions/setup-node - either using it, or doing something similar. Though TBH, that seems like unnecessary overhead here - node12 should be fine.

I'll revert to 12 for the PR, and maybe investigate options based on performance.

**Comment by mchenryc** [comment]
Tests pass [in my fork](https://github.com/mchenryc/spotbugs-github-action/runs/3644841534), but because the action posts results, the PR fails, which is apparently a known issue:
* https://github.com/actions/first-interaction/issues/10
* https://github.community/t/github-actions-are-severely-limited-on-prs

**Comment by jwgmeligmeyling** [comment]
Yes, I have been thinking about a proper way to deal with this issue for a while.

I'll just merge your PR, thanks for the contribution! 👍🏻

**Comment by jwgmeligmeyling** [comment]
After giving it a bit of a thought, I think I should create a Github Token for this repository and set it through `inputs.token`.

That way also PR's from forks have write permission: https://github.community/t/github-actions-are-severely-limited-on-prs/18179/2 (comment from the thread you've posted)

**Comment by mchenryc** [comment]
Seems that would give full permission to the repo, not just permission to comment / post results. It would be a malicious case, but someone could add actions in `.github/actions/...` that could modify the repo directly. Though first-time contributors need to be approved before workflows run in their PRs, it sounds like workflows run automatically for contributors thereafter. This seems to give full write access to anyone who's contributed to the project (though it require intentional hijinks). I expect that is why GH does not recommend it.

@jwgmeligmeyling near as I can tell, the "official" solution by GH is `pull_request_target` which allows fork pull requests to have write permission _if_ you have given the requester [collaborator access](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-user-account-settings/permission-levels-for-a-user-account-repository#collaborator-access-for-a-repository-owned-by-a-user-account). See [this comment](https://github.community/t/github-actions-are-severely-limited-on-prs/18179/17) from the same thread.

Introductory description here: https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/. Essentially, you still give write access but with explicit per-user permissions that can be revoked later, rather than a token good for anyone.

**Pull request merged**